### PR TITLE
bugfix: CLDSRV-280 add check for object put retention for object put

### PR DESCRIPTION
**Pull request by miniscruff**

### CODE REVIEW

**Review by alexanderchan-scality — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

File: lib/api/apiUtils/authorization/prepareRequestContexts.js [hunk 1]
```diff
@@ -118,6 +118,12 @@ function prepareRequestContexts(apiMethod, request, sourceBucket,
               generateRequestContext('objectPutACL');
             requestContexts.push(putAclRequestContext);
         }
+
+        if (request.headers['x-amz-object-lock-mode']) {
+            const putObjectLockRequestConext =
+              generateRequestContext('objectPutRetention');
+            requestContexts.push(putObjectLockRequestConext);
```
Comment (ghost) [inline lib/api/apiUtils/authorization/prepareRequestContexts.js]: Small typo:

```suggestion
            const putObjectLockRequestContext =
              generateRequestContext('objectPutRetention');
            requestContexts.push(putObjectLockRequestContext);
```

Comment (miniscruff) [reply lib/api/apiUtils/authorization/prepareRequestContexts.js]: oof, good catch, maybe the queue is stuck and I can merge this as well...

**Comment by miniscruff** [comment]
@bert-e ping

**Comment by miniscruff** [comment]
@bert-e help

**Comment by miniscruff** [comment]
@bert-e wait

**Pull request merged**